src/coordinator/storage/local/storage_test.go

```diff
@@ -85,32 +122,95 @@ func newWriteQuery() *storage.WriteQuery {
 	}
 }
 
-func setupLocalWrite(t *testing.T) storage.Storage {
-	ctrl := gomock.NewController(t)
-	store, session := setup(ctrl)
-	session.EXPECT().WriteTagged(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+func setupLocalWrite(t *testing.T, ctrl *gomock.Controller) storage.Storage {
+	store, sessions := setup(t, ctrl)
+	session := sessions.unaggregated1MonthRetention
+	session.EXPECT().WriteTagged(gomock.Any(), gomock.Any(), gomock.Any(),
+		gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
 	return store
 }
 
 func TestLocalWriteEmpty(t *testing.T) {
-	store := setupLocalWrite(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store := setupLocalWrite(t, ctrl)
 	err := store.Write(context.TODO(), nil)
 	assert.Error(t, err)
 }
 
 func TestLocalWriteSuccess(t *testing.T) {
-	store := setupLocalWrite(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store := setupLocalWrite(t, ctrl)
 	writeQuery := newWriteQuery()
 	err := store.Write(context.TODO(), writeQuery)
 	assert.NoError(t, err)
 	assert.NoError(t, store.Close())
 }
 
+func TestLocalWriteAggregatedNoClusterNamespaceError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store, _ := setup(t, ctrl)
+	writeQuery := newWriteQuery()
+	// Use unsupported retention/resolution
+	writeQuery.Attributes = storage.Attributes{
+		MetricsType: storage.AggregatedMetricsType,
+		Retention:   1234,
+		Resolution:  5678,
+	}
+	err := store.Write(context.TODO(), writeQuery)
+	assert.Error(t, err)
+	assert.True(t, strings.Contains(err.Error(), "no configured cluster namespace"),
```

Comment:
I think you can do something like `assert.EqualError`? Might be a bit iffy if you don't have the exact error string we're expecting though

Reply:
Yeah, that's why I prefer the Contains... check.

Reply:
In this case, could it be worth at some point pulling out a bunch of test utilities like this into a custom  module to allow us to reuse them?

Reply:
Possibly, although it's pretty simply so I'm fine to also just rely on the standard library and `stretchr` without adding another package into the mix for now.